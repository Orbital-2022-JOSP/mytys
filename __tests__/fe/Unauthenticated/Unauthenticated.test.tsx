import { render } from '@testing-library/react';
import { Unauthenticated } from '../../../components/Unauthenticated/Unauthenticated';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Unauthenticated Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<Unauthenticated />);

            expect(container).toMatchSnapshot()
        })
})

