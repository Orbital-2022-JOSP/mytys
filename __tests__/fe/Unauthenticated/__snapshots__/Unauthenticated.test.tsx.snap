// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Unauthenticated Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <div
      class="max-w-xl sm:mx-auto lg:max-w-2xl"
    >
      <div
        class="flex flex-col mb-16 sm:text-center sm:mb-0"
      >
        <a
          class="mb-6 sm:mx-auto"
          href="/"
        >
          <div
            class="flex items-center justify-center w-12 h-12 rounded-full bg-blue-50"
          >
            <svg
              class="w-10 h-10 text-light-blue-accent-400"
              stroke="currentColor"
              viewBox="0 0 52 52"
            >
              <polygon
                fill="none"
                points="29 13 14 29 25 29 23 39 38 23 27 23"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
              />
            </svg>
          </div>
        </a>
        <div
          class="max-w-xl mb-10 md:mx-auto sm:text-center lg:max-w-2xl md:mb-12"
        >
          <h2
            class="max-w-lg mb-6 font-sans text-3xl font-bold leading-none tracking-tight text-gray-900 sm:text-4xl md:mx-auto"
          >
            <span
              class="relative inline-block"
            >
              <svg
                class="absolute top-0 left-0 z-0 hidden w-32 -mt-8 -ml-20 text-blue-gray-100 lg:w-32 lg:-ml-28 lg:-mt-10 sm:block"
                fill="currentColor"
                viewBox="0 0 52 24"
              >
                <defs>
                  <pattern
                    height=".30"
                    id="e77df901-b9d7-4b9b-822e-16b2d410795b"
                    width=".135"
                    x="0"
                    y="0"
                  >
                    <circle
                      cx="1"
                      cy="1"
                      r=".7"
                    />
                  </pattern>
                </defs>
                <rect
                  fill="url(#e77df901-b9d7-4b9b-822e-16b2d410795b)"
                  height="24"
                  width="52"
                />
              </svg>
            </span>
            Unauthenticated
          </h2>
          <p
            class="text-base text-gray-700 md:text-lg"
          >
            Looks like you're not signed in, please do so to view this content!
          </p>
        </div>
        <div>
          <button
            class="inline-flex items-center justify-center h-12 px-6 font-medium tracking-wide text-white transition duration-200 rounded shadow-md bg-light-blue-accent-400 hover:bg-light-blue-accent-700 focus:shadow-outline focus:outline-none"
          >
            Log In
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
